plugins/github-pull-requests-board/src/api/useGetPullRequestsFromRepository.ts

```diff
@@ -23,32 +23,106 @@ export const useGetPullRequestsFromRepository = () => {
     useOctokitGraphQl<GraphQlPullRequests<PullRequestsNumber[]>>();
 
   const fn = React.useRef(
-    async (repo: string): Promise<PullRequestsNumber[]> => {
+    async (
+      repo: string,
+      defaultLimit?: number,
+    ): Promise<PullRequestsNumber[]> => {
       const [organisation, repositoryName] = repo.split('/');
 
-      const { repository } = await graphql(
-        `
-          query ($name: String!, $owner: String!) {
-            repository(name: $name, owner: $owner) {
-              pullRequests(states: OPEN, first: 10) {
-                edges {
-                  node {
-                    number
-                  }
-                }
-              }
-            }
-          }
-        `,
-        {
-          name: repositoryName,
-          owner: organisation,
-        },
-      );
+      if (defaultLimit) {
```

Comment:
Thinking about this, could we not have a constant for the default limit that is 10 as that matches the current state and then say if not provided fall back to the constant value? This is how we handle this for the Azure DevOps plugin and it works well and would save us this if block.

```ts
const FIRST_PULL_REQUEST_LIMIT = 10
const firstPullRequestLimit = defaultLimit ?? FIRST_PULL_REQUEST_LIMIT
```